### PR TITLE
Add link list properties to nodes

### PR DESCRIPTION
**Pull request by trueadm**
This PR introduces the node properties required for the work to move Lexical's internal structure to a doubly linked list in the future. Note: we don't actually do anything with these properties in this PR.

### CODE REVIEW

File: packages/lexical/src/nodes/LexicalElementNode.ts [hunk 1]
```diff
@@ -55,6 +55,10 @@ export class ElementNode extends LexicalNode {
   /** @internal */
   __children: Array<NodeKey>;
   /** @internal */
+  __first: null | NodeKey;
+  /** @internal */
+  __last: null | NodeKey;
```
Comment (acywatson) [inline packages/lexical/src/nodes/LexicalElementNode.ts:60]: Consider adding “__size” here as well

Comment (trueadm) [reply packages/lexical/src/nodes/LexicalElementNode.ts:60]: Nice one.

**Comment by acywatson** [comment]
I was wondering if it'd make more sense to develop this feature on a branch, as opposed to merging it in piecemeal. Makes it harder for others to understand what has and hasn't been done. Will leave it up to you though.

**Review by acywatson — APPROVED** [review]

**Comment by trueadm** [comment]
@acywatson I wanted to get this in just to simply ensure we have the properties on the nodes. Having them there means that people might not assume they're not there and do breaking change stuff with Lexical's internal node structures (not that they should).

**Pull request merged**